### PR TITLE
6 ➡️ 7 (main)

### PR DESCRIPTION
**Pull request by chapulina**
# 6 ➡️ 7

* Port `ign-gazebo6` to `main`
* Branch comparison: https://github.com/ignitionrobotics/ign-gazebo/compare/main...ign-gazebo6
* Requires https://github.com/ignitionrobotics/ign-msgs/pull/244

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

**Comment by mjcarroll** [comment]
`focal` is busted missing `dataframe`.

**Comment by chapulina** [comment]
Yup, this needs https://github.com/ignitionrobotics/ign-msgs/pull/244 to go in first and then some nightlies

**Comment by chapulina** [comment]
I'm not sure what's up with the failure on Jenkins Ubuntu:

```
Errors were encountered while processing:
 /tmp/apt-dpkg-install-M4iyFG/023-libignition-common5-core-dev_4.999.999+nightly+git20220420+1r1b1ad37b31be684b3679d8b92cca957addeaab2f-1~focal_amd64.deb
```

GitHub actions was able to install all dependencies just fine, and I can't reproduce the issue locally. I've already tried cleaning the workspace. Maybe there's some other caching going on? CC @j-rivero

**Comment by chapulina** [comment]
> failure on Jenkins Ubuntu

Should be fixed by

* https://github.com/ignition-release/ign-common5-release/pull/10

There are `ign-common5` nightlies building, then we can retrigger CI here:

https://build.osrfoundation.org/job/ign-common5-debbuilder/

**Comment by chapulina** [comment]
CI status:

* Actions Focal: Only a pre-existing test failure:
    * `INTEGRATION_scene_broadcaster_system`
* Action Jammy: new test failures. Jammy CI is being introduced by this branch, so I think we should merge it like this and address the test failures in a separate PR.
    ```
  	103 - INTEGRATION_follow_actor_system (SEGFAULT)
  	139 - INTEGRATION_model_photo_shoot_default_joints (SEGFAULT)
  	141 - INTEGRATION_model_photo_shoot_random_joints (SEGFAULT)
  	173 - INTEGRATION_scene_broadcaster_system (Failed)
  	209 - testFixture_TEST.py (Subprocess aborted)
    ```
* Windows: only warnings, as expected
* Homebrew: pre-existing failures:
    * `INTEGRATION_scene_broadcaster_system`
    * `UNIT_SimulationRunner_TEST`
* Jenkins Ubuntu: Infra is broken right now with `nvidia-container-cli: initialization error: nvml error: driver/library version mismatch: unknown.`

File: .github/ci/packages-jammy.apt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+libdart-collision-ode-dev
+libdart-dev
+libdart-external-ikfast-dev
+libdart-external-odelcpsolver-dev
+libdart-utils-urdf-dev
+libogre-next-dev
+python3-ignition-math6
```
Comment (methylDragon) [inline .github/ci/packages-jammy.apt:7]: Should this be math7?

Comment (chapulina) [reply .github/ci/packages-jammy.apt:7]: yes! Thanks!

**Review by adityapande-1995 — COMMENTED** [review]
In order for https://github.com/ignitionrobotics/ign-gazebo/pull/1446/commits/2e8892ae552f73e8a932d797a1449eadb84a0e38 to work,

https://github.com/ignitionrobotics/ign-sensors/pull/186 and 
https://github.com/ignitionrobotics/ign-sensors/pull/212 would also need to be forward ported from ``ign-sensors``

**Comment by chapulina** [comment]
> In order for https://github.com/ignitionrobotics/ign-gazebo/commit/2e8892ae552f73e8a932d797a1449eadb84a0e38 to work,

Thanks, I think those have already been ported on these PRs

* https://github.com/ignitionrobotics/ign-sensors/pull/209
* https://github.com/ignitionrobotics/ign-sensors/pull/214

A convenient way to check what hasn't been ported yet is to click on the links on [this table](https://github.com/ignitionrobotics/docs/blob/master/tools/branch_comparisons.md).

File: .github/ci/packages-jammy.apt (hunk 1 above)
Comment (chapulina) [reply .github/ci/packages-jammy.apt]: [e399431](https://github.com/ignitionrobotics/ign-gazebo/pull/1446/commits/e399431488445098a8b889021fa6a0705ad4a403)

**Review by adityapande-1995 — APPROVED** [review]
Looks good, hopefully no errors other than these : https://github.com/ignitionrobotics/ign-gazebo/pull/1446#issuecomment-1107003792

**Pull request merged**